libraries/redcore/database/driver.php

```diff
@@ -248,5 +247,203 @@ public static function deleteInstances()
 		{
 			self::$instances = null;
 		}
+
+		/**
+		 * Splits a string of multiple queries into an array of individual queries.
+		 *
+		 * @param   string  $sql  Input SQL string with which to split into individual queries.
+		 *
+		 * @return  array  The queries from the input string separated into an array.
+		 *
+		 * @since   11.1
+		 */
+		public static function splitSql($sql)
+		{
+			$start = 0;
+			$open = false;
+			$char = '';
+			$end = strlen($sql);
+			$queries = array();
+			$delimiter = ';';
+			$line = '';
+			$latest = '';
+
+			for ($i = 0; $i < $end; $i++)
+			{
+				$current = substr($sql, $i, 1);
+				$latest .= $current;
+
+				if (strlen($latest) > strlen($delimiter))
+				{
+					$latest = substr($latest, 1);
```

Comment:
shouldn't if (strlen($latest) > strlen($delimiter)) always be true if you add up $latest .= $current;
why not always just use $latest = $current; then you will not have to use $latest = substr($latest, 1);